elasticsearch/src/main/java/com/tcdi/zombodb/query_parser/IndexLinkOptimizer.java

```diff
@@ -32,7 +32,7 @@
     private final ASTQueryTree tree;
     private final IndexMetadataManager metadataManager;
 
-    private Set<ASTIndexLink> usedIndexes = new HashSet<>();
+    private Map<ASTIndexLink, Boolean> usedIndexes = new IdentityHashMap<>();
```

Comment:
What's the purpose of this change?  This class used to use `IdentityHashMap`, but was changed to use a `HashSet` because IdentityHashMap has the wrong semantics.  

This needs to be changed back.